### PR TITLE
Add amqp meta

### PR DESCRIPTION
**Pull request by b1rdex**
I wonder should this be moved to `amqp/.phpstorm.meta.php` file?

### CODE REVIEW

**Comment by MaXal** [comment]
@b1rdex It's fine as it is since we support only `.phpstorm.meta.php` file and a set of files inside `.phpstorm.meta`. So `amp/.phpstorm.meta.php` unfortunately won't work.

**Comment by Leneshka-jb** [comment]
@MaXal @b1rdex but we would like to have such approach working, so I'll try to support it and merge then

File: meta/.phpstorm.meta.php [hunk 1]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
     expectedArguments(\yaml_emit(), 1, YAML_ANY_ENCODING, YAML_UTF8_ENCODING, YAML_UTF16LE_ENCODING, YAML_UTF16BE_ENCODING);
     expectedArguments(\yaml_emit(), 2, YAML_ANY_BREAK, YAML_CR_BREAK, YAML_LN_BREAK, YAML_CRLN_BREAK);
 
+    expectedArguments(\AMQPExchange::delete(), 1, AMQP_NOPARAM, AMQP_IFUNUSED);
+    expectedArguments(\AMQPExchange::publish(), 2, AMQP_NOPARAM | AMQP_MANDATORY | AMQP_IMMEDIATE);
+    expectedArguments(\AMQPExchange::setFlags(), 0, AMQP_PASSIVE | AMQP_DURABLE);
```
Comment (Leneshka-jb) [inline meta/.phpstorm.meta.php]: Looks to me `AMQP_AUTODELETE` and `AMQP_INTERNAL` are also useful here, same for `getFlags`

File: meta/.phpstorm.meta.php [hunk 2]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
...
+    expectedReturnValues(\AMQPExchange::getFlags(), AMQP_PASSIVE | AMQP_DURABLE);
+
+    expectedArguments(\AMQPQueue::ack(), 1, AMQP_NOPARAM, AMQP_MULTIPLE);
+    expectedArguments(\AMQPQueue::consume(), 1, AMQP_NOPARAM | AMQP_AUTOACK | AMQP_JUST_CONSUME);
```
Comment (Leneshka-jb) [inline meta/.phpstorm.meta.php]: Also `AMQP_NOLOCAL`

File: meta/.phpstorm.meta.php [hunk 3]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
...
+    expectedArguments(\AMQPQueue::delete(), 0, AMQP_NOPARAM, AMQP_IFUNUSED);
```
Comment (Leneshka-jb) [inline meta/.phpstorm.meta.php]: Also `AMQP_IFEMPTY`. Could you please explain, why are they exclusive here?

File: meta/.phpstorm.meta.php [hunk 4]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
...
+    expectedArguments(\AMQPQueue::get(), 0, AMQP_NOPARAM | AMQP_AUTOACK);
```
Comment (Leneshka-jb) [inline meta/.phpstorm.meta.php]: Looks `AMQP_NOPARAM, AMQP_AUTOACK` for me, for the only usage is `(AMQP_AUTOACK & flags) ? 1 : 0`

File: meta/.phpstorm.meta.php [hunk 5]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
...
+    expectedArguments(\AMQPQueue::nack(), 1, AMQP_NOPARAM, AMQP_REQUEUE, AMQP_MULTIPLE);
```
Comment (Leneshka-jb) [inline meta/.phpstorm.meta.php]: Could you please explain why the options are exclusive here?

**Comment by Leneshka-jb** [comment]
I decided to merge improvement now and then refactor after the feature implementation.

According to sources, there are some additional possible values. Also I'm not sure about exclusiveness of some options and would be happy to hear some comments.

File: meta/.phpstorm.meta.php (hunk 2 above)
Comment (b1rdex) [inline meta/.phpstorm.meta.php]:
```suggestion
    expectedArguments(\AMQPQueue::consume(), 1, AMQP_NOPARAM | AMQP_AUTOACK | AMQP_JUST_CONSUME | AMQP_NOLOCAL);
```

File: meta/.phpstorm.meta.php (hunk 1 above)
Comment (b1rdex) [inline meta/.phpstorm.meta.php]:
```suggestion
    expectedArguments(\AMQPExchange::setFlags(), 0, AMQP_PASSIVE | AMQP_DURABLE | AMQP_AUTODELETE | AMQP_INTERNAL);
```

File: meta/.phpstorm.meta.php [hunk 6]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
     expectedArguments(\yaml_emit(), 1, YAML_ANY_ENCODING, YAML_UTF8_ENCODING, YAML_UTF16LE_ENCODING, YAML_UTF16BE_ENCODING);
     expectedArguments(\yaml_emit(), 2, YAML_ANY_BREAK, YAML_CR_BREAK, YAML_LN_BREAK, YAML_CRLN_BREAK);
 
+    expectedArguments(\AMQPExchange::delete(), 1, AMQP_NOPARAM, AMQP_IFUNUSED);
+    expectedArguments(\AMQPExchange::publish(), 2, AMQP_NOPARAM | AMQP_MANDATORY | AMQP_IMMEDIATE);
+    expectedArguments(\AMQPExchange::setFlags(), 0, AMQP_PASSIVE | AMQP_DURABLE);
+    expectedReturnValues(\AMQPExchange::getFlags(), AMQP_PASSIVE | AMQP_DURABLE);
```
Comment (b1rdex) [inline meta/.phpstorm.meta.php]:
```suggestion
    expectedReturnValues(\AMQPExchange::getFlags(), AMQP_PASSIVE | AMQP_DURABLE | AMQP_AUTODELETE | AMQP_INTERNAL);
```

File: meta/.phpstorm.meta.php [hunk 7]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
...
+
+    expectedArguments(\AMQPQueue::ack(), 1, AMQP_NOPARAM, AMQP_MULTIPLE);
+    expectedArguments(\AMQPQueue::consume(), 1, AMQP_NOPARAM | AMQP_AUTOACK | AMQP_JUST_CONSUME | AMQP_NOLOCAL);
+    expectedArguments(\AMQPQueue::delete(), 0, AMQP_NOPARAM, AMQP_IFUNUSED);
+    expectedArguments(\AMQPQueue::get(), 0, AMQP_NOPARAM | AMQP_AUTOACK);
```
Comment (b1rdex) [inline meta/.phpstorm.meta.php]:
```suggestion
    expectedArguments(\AMQPQueue::get(), 0, AMQP_NOPARAM, AMQP_AUTOACK);
```

File: meta/.phpstorm.meta.php [hunk 8]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
...
     expectedArguments(\yaml_emit(), 2, YAML_ANY_BREAK, YAML_CR_BREAK, YAML_LN_BREAK, YAML_CRLN_BREAK);
 
+    expectedArguments(\AMQPExchange::delete(), 1, AMQP_NOPARAM, AMQP_IFUNUSED);
+    expectedArguments(\AMQPExchange::publish(), 2, AMQP_NOPARAM | AMQP_MANDATORY | AMQP_IMMEDIATE);
+    expectedArguments(\AMQPExchange::setFlags(), 0, AMQP_PASSIVE | AMQP_DURABLE);
+    expectedReturnValues(\AMQPExchange::getFlags(), AMQP_PASSIVE | AMQP_DURABLE);
+
+    expectedArguments(\AMQPQueue::ack(), 1, AMQP_NOPARAM, AMQP_MULTIPLE);
+    expectedArguments(\AMQPQueue::consume(), 1, AMQP_NOPARAM | AMQP_AUTOACK | AMQP_JUST_CONSUME | AMQP_NOLOCAL);
+    expectedArguments(\AMQPQueue::delete(), 0, AMQP_NOPARAM, AMQP_IFUNUSED);
```
Comment (b1rdex) [inline meta/.phpstorm.meta.php]:
```suggestion
    expectedArguments(\AMQPQueue::delete(), 0, AMQP_NOPARAM | AMQP_IFUNUSED | AMQP_IFEMPTY);
```

File: meta/.phpstorm.meta.php [hunk 9]
```diff
@@ -505,6 +505,20 @@ function expectedReturnValues($functionReference, $values) {
...
+    expectedArguments(\AMQPQueue::get(), 0, AMQP_NOPARAM | AMQP_AUTOACK);
+    expectedArguments(\AMQPQueue::nack(), 1, AMQP_NOPARAM, AMQP_REQUEUE, AMQP_MULTIPLE);
```
Comment (b1rdex) [inline meta/.phpstorm.meta.php]:
```suggestion
    expectedArguments(\AMQPQueue::nack(), 1, AMQP_NOPARAM | AMQP_REQUEUE | AMQP_MULTIPLE);
```

**Comment by b1rdex** [comment]
I've fixed the issues.

**Pull request merged**